hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -212,9 +212,7 @@ public int getDefaultDecimalPrecision() {
 
 	@Override
 	public int getPreferredSqlTypeCodeForBoolean() {
-		return getVersion().isBefore( 10, 7 )
-				? Types.SMALLINT
-				: Types.BOOLEAN;
+		return Types.BOOLEAN;
 	}
```

Comment:
I think you can just remove the method as the super implementation will do the right thing